src/transports/http.c

```diff
@@ -718,10 +735,177 @@ static int stream_connect(
 	return error;
 }
 
+static int gen_connect_req(git_buf *buf, http_subtransport *t)
+{
+	git_buf_printf(buf, "CONNECT %s:%s HTTP/1.1\r\n",
+		t->server.url.host, t->server.url.port);
+
+	git_buf_puts(buf, "User-Agent: ");
+	git_http__user_agent(buf);
+	git_buf_puts(buf, "\r\n");
+
+	git_buf_printf(buf, "Host: %s\r\n", t->proxy.url.host);
+
+	if (apply_credentials(buf, &t->proxy, AUTH_HEADER_PROXY) < 0)
+		return -1;
+
+	git_buf_puts(buf, "\r\n");
+
+	return git_buf_oom(buf) ? -1 : 0;
+}
+
+static int proxy_headers_complete(http_parser *parser)
+{
+    parser_context *ctx = (parser_context *) parser->data;
+    http_subtransport *t = ctx->t;
+    int proxy_auth_types = 0;
+
+    /* Enforce a reasonable cap on the number of replays */
+    if (++t->replay_count >= GIT_HTTP_REPLAY_MAX) {
```

Comment:
Again, this caps us to 6 replays due to the pre-increment

Reply:
Indeed - this was to emulate https://github.com/libgit2/libgit2/blob/master/src/transports/winhttp.c#L932 but I agree that it's annoying.  I'll fix.